src/nvim/ex_getln.c

```diff
@@ -347,6 +347,13 @@ static uint8_t *command_line_enter(int firstc, long count, int indent)
     redrawcmd();
   }
 
+  // redraw the statusline for statuslines that display the current mode
+  // using the mode() function.
+  if (KeyTyped) {
+    curwin->w_redr_status = true;
```

Comment:
Technically this is needed for all windows, not only `curwin`. But if you don't care I don't care :)